File: chapter4/reversing_an_array.rb
def reverse_array(ary)
  ary.each_with_object([]) { _2.unshift(_1) }
end

p reverse_array([1, 2, 3])

def reverse_array_in_place(ary)
  (ary.size / 2).times do |i|
    ary[i], ary[-1 - i] = ary[-1 - i], ary[i]
  end
  ary
end

p reverse_array_in_place([1, 2, 3, 4, 5, 6])

File: chapter4/reversing_an_array.js
const reverseArray = arr => {
  const reversedArray = [];
  arr.forEach(element => reversedArray.unshift(element));
  return reversedArray;
}

console.log(reverseArray([1, 2, 3, 4]));

const reverseArrayInPlace = arr => {
  for (let i = 0; i < Math.floor(arr.length / 2); i++) {
    let temp = arr[i];
    arr[i] = arr[arr.length - 1 - i];
    arr[arr.length - 1 - i] = temp;
  }
  return arr;
}

console.log(reverseArrayInPlace([1, 2, 3, 4]));
